syscall/syscall.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebiten Authors
+
+//go:build darwin
+// +build darwin
+
+package syscall
+
+import "unsafe"
+
+const maxArgs = 9
+
+func SyscallN(fn uintptr, args ...uintptr) (r1, r2, err uintptr) {
+	if len(args) < maxArgs {
+		// add padding so there is no out-of-bounds slicing
+		var tmp = make([]uintptr, maxArgs)
+		copy(tmp, args)
+		args = tmp
+	}
+	switch len(args) {
```

Comment:
Isn't `len(args)` always 9?

Reply:
Yes it is. I feel dumb now. I just tested and only having the 9 argument version works properly on all the tests.